diff --git a/android/build/cmake/android.cmake b/android/build/cmake/android.cmake
--- a/android/build/cmake/android.cmake
+++ b/android/build/cmake/android.cmake
@@ -89,11 +89,17 @@ function(android_compile_for_host EXE SOURCE OUT_PATH)
         DOWNLOAD_COMMAND ""
         SOURCE_DIR ${SOURCE}
         CMAKE_ARGS
           ${ANDROID_HOST_TOOLCHAIN_CMAKE_ARGS}
+          "-DCMAKE_MODULE_PATH=${ANDROID_QEMU2_TOP_DIR}/android/build/cmake"
+          "-DANDROID_QEMU2_TOP_DIR=${ANDROID_QEMU2_TOP_DIR}"
+          "-DANDROID_HOST_TAG=${ANDROID_HOST_TAG}"
+          "-DANDROID_TARGET_TAG=${ANDROID_HOST_TAG}"
+          "-DLINUX_X86_64=TRUE"
+          "-DOPTION_SYSTEM_HOST_TOOLCHAIN=TRUE"
           "-DBREAKPAD_API_KEY=${BREAKPAD_API_KEY}"
           "-DBREAKPAD_API_URL=${BREAKPAD_API_URL}"
           "-DOPTION_TEST_LOGS=${OPTION_TEST_LOGS}"
           "-DIN_CROSS_COMPILE=TRUE"
           "-DOPTION_BAZEL=TRUE"
           "-DANDROID_SYMBOL_DIR=${ANDROID_SYMBOL_DIR}"
           "-DPython_EXECUTABLE=${Python_EXECUTABLE}"
