diff --git a/minidump/minidump_extensions.h b/minidump/minidump_extensions.h
--- a/minidump/minidump_extensions.h
+++ b/minidump/minidump_extensions.h
@@ -34,10 +34,10 @@
 #pragma warning(push)
 #pragma warning(disable: 4200)

-#define PACKED
+#define PACKED alignas(4)
 #pragma pack(push, 1)
 #else
-#define PACKED __attribute__((packed))
+#define PACKED __attribute__((packed, aligned(4)))
 #endif  // COMPILER_MSVC

 namespace crashpad {
@@ -116,7 +116,7 @@
 //!     file.
 //!
 //! \sa MINIDUMP_STRING
-struct alignas(4) PACKED MinidumpUTF8String {
+struct PACKED MinidumpUTF8String {
   // The field names do not conform to typical style, they match the names used
   // in MINIDUMP_STRING. This makes it easier to operate on MINIDUMP_STRING (for
   // UTF-16 strings) and MinidumpUTF8String using templates.
@@ -135,7 +135,7 @@
 //! \brief A variable-length array of bytes carried within a minidump file.
 //!     The data have no intrinsic type and should be interpreted according
 //!     to their referencing context.
-struct alignas(4) PACKED MinidumpByteArray {
+struct PACKED MinidumpByteArray {
   //! \brief The length of the #data field.
   uint32_t length;

@@ -273,7 +273,7 @@
 };

 //! \brief A list of ::RVA pointers.
-struct alignas(4) PACKED MinidumpRVAList {
+struct PACKED MinidumpRVAList {
   //! \brief The number of children present in the #children array.
   uint32_t count;

@@ -282,7 +282,7 @@
 };

 //! \brief A key-value pair.
-struct alignas(4) PACKED MinidumpSimpleStringDictionaryEntry {
+struct PACKED MinidumpSimpleStringDictionaryEntry {
   //! \brief ::RVA of a MinidumpUTF8String containing the key of a key-value
   //!     pair.
   RVA key;
@@ -293,7 +293,7 @@
 };

 //! \brief A list of key-value pairs.
-struct alignas(4) PACKED MinidumpSimpleStringDictionary {
+struct PACKED MinidumpSimpleStringDictionary {
   //! \brief The number of key-value pairs present.
   uint32_t count;

@@ -302,7 +302,7 @@
 };

 //! \brief A typed annotation object.
-struct alignas(4) PACKED MinidumpAnnotation {
+struct PACKED MinidumpAnnotation {
   //! \brief ::RVA of a MinidumpUTF8String containing the name of the
   //!     annotation.
   RVA name;
@@ -319,7 +319,7 @@
 };

 //! \brief A list of annotation objects.
-struct alignas(4) PACKED MinidumpAnnotationList {
+struct PACKED MinidumpAnnotationList {
   //! \brief The number of annotation objects present.
   uint32_t count;

@@ -342,7 +342,7 @@
 //! fields are valid or not.
 //!
 //! \sa MinidumpModuleCrashpadInfoList
-struct alignas(4) PACKED MinidumpModuleCrashpadInfo {
+struct PACKED MinidumpModuleCrashpadInfo {
   //! \brief The structure’s currently-defined version number.
   //!
   //! \sa version
@@ -392,7 +392,7 @@
 //! \brief A link between a MINIDUMP_MODULE structure and additional
 //!     Crashpad-specific information about a module carried within a minidump
 //!     file.
-struct alignas(4) PACKED MinidumpModuleCrashpadInfoLink {
+struct PACKED MinidumpModuleCrashpadInfoLink {
   //! \brief A link to a MINIDUMP_MODULE structure in the module list stream.
   //!
   //! This field is an index into MINIDUMP_MODULE_LIST::Modules. This field’s
@@ -419,7 +419,7 @@
 //! structure carried within the minidump file will necessarily have
 //! Crashpad-specific information provided by a MinidumpModuleCrashpadInfo
 //! structure.
-struct alignas(4) PACKED MinidumpModuleCrashpadInfoList {
+struct PACKED MinidumpModuleCrashpadInfoList {
   //! \brief The number of children present in the #modules array.
   uint32_t count;

@@ -438,7 +438,7 @@
 //! structure. Revise #kVersion and document each field’s validity based on
 //! #version, so that newer parsers will be able to determine whether the added
 //! fields are valid or not.
-struct alignas(4) PACKED MinidumpCrashpadInfo {
+struct PACKED MinidumpCrashpadInfo {
   // UUID has a constructor, which makes it non-POD, which makes this structure
   // non-POD. In order for the default constructor to zero-initialize other
   // members, an explicit constructor must be provided.
